src/ray/core_worker/core_worker.cc

```diff
@@ -2103,6 +2086,13 @@ Status CoreWorker::Wait(const std::vector<ObjectID> &ids,
           worker_context_,
           &ready));
     }
+  } else {
+    for (const auto &object_id : plasma_object_ids) {
```

Comment:
please add a comment for this code block and the `fetch_local` one above explaining the divergence in behavior. the behavior is non-obvious to an uninformed reader